### PR TITLE
Changed build.gradle to functioning version + Added sign-data command…

### PR DESCRIPTION
**Pull request by danilobispo**
… to library

### CODE REVIEW

File: src/main/java/com/lacunasoftware/pkiexpress/Authentication.java [hunk 1]
```diff
@@ -77,6 +78,10 @@ public void setSignature(String signatureBase64) {
 		this.signature = signatureBase64;
 	}
 
+	public void setSignature(byte[] signature) {
+		this.signature = new String(signature);
+	}
+
```
Comment (medisco) [inline src/main/java/com/lacunasoftware/pkiexpress/Authentication.java:80]: Here, you are converting a byte[] into a hexadecimal string, it's not what PKI Express supports. You should convert this byte[] into a base64 string. Something like the code below:

```java
this.signature = Base64.getEncoder().encodeToString(signature);
```

File: src/main/java/com/lacunasoftware/pkiexpress/CertificateExplorer.java [hunk 2]
```diff
@@ -66,44 +68,54 @@ public void setValidate(boolean validate) {
...
 	public CertificateExplorerResult open() throws IOException {
-		if (certificatePath == null) {
-			throw new RuntimeException("The certificate file was not set");
+		if (certificatePath == null && trustServiceSession == null) {
+			throw new RuntimeException("The certificate file or the certificate session were not set");
 		}
-
+		
 		List<String> args = new ArrayList<String>();
 		args.add("--file");
```
Comment (medisco) [inline src/main/java/com/lacunasoftware/pkiexpress/CertificateExplorer.java]: Here, the new logic should be like this:
- if the field `certificatePath` is filled, add the flag with field: `--file <certificatePath>`
- if the field `trustServiceSession` is filled, add the flag with field: `--trust-service-session <session>`
The developer should pass either a file path or a session for PKIE to open the desired certificate

TLDR: You should do this:
```java
List<String> args = new ArrayList<String>();
if  (certificatePath != null) {
   args.add("--file");
   args.add(certificatePath.toString());
}

if (trustServiceSession != null) {
   args.add("--trust-service-session"):
   args.add(trustServiceSession);
}
```

File: src/main/java/com/lacunasoftware/pkiexpress/CertificateExplorer.java [hunk 3]
```diff
@@ -118,6 +130,10 @@ public CertificateExplorerResult open() throws IOException {
 			this.versionManager.requireVersion(new Version("1.22"));
 		}
 
+		if(trustServiceSession != null){
+			args.add("--trust-service-session");
```
Comment (medisco) [inline src/main/java/com/lacunasoftware/pkiexpress/CertificateExplorer.java:137]: I mentioned in the comment above, that is field trustServiceSession is not being passed to `args` to be used in the command

File: src/main/java/com/lacunasoftware/pkiexpress/DataSigner.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        return this.toSignData;
+    }
+    
+    public void setToSignData(String toSignData) {
+        this.toSignData = toSignData;
+    }
+
+    public DataSigner(PkiExpressConfig config) {
+		super(config);
+	}
```
Comment (medisco) [inline src/main/java/com/lacunasoftware/pkiexpress/DataSigner.java]: There are some inconsistencies in the identation of this file (probably a mix of tabs and spaces). Please use only tabs in this project

**Review by medisco — CHANGES_REQUESTED** [review]

**Review by medisco — APPROVED** [review]
LGTM!
